### PR TITLE
Implemented file settings (p2vp_settings.ini)

### PR DESCRIPTION
**Pull request by lilmayu**
Hello once again,

In this PR I have implemented settings loading from .ini file to allow tech savvy users (or external programs) set-up the address and other various settings.

## Basics
User can create file `p2vp_settings.ini` in the same directory as `th19.exe` with various settings to be loaded on startup.

## Implemented settings
`host_ip` - The host IP that user enters in the text box
`delay_compensation` - The delay compensation number
`host_port` - host port
`guest_port` - guest port
`log_enabled` - enables / disables the log (same behaviour as the check box in the Advanced Settings)
`show_console` - if enabled, shows the console immediately on the start (after the settings are loaded)

The `p2vp_settings.ini` looks like this
```ini
[Connection]
host_ip=[::1]:18000
delay_compensation=4
host_port=18000
guest_port=18001

[Utils]
log_enabled=false
show_console=false
```

## 3rd party library to load .ini files
I have used open source library named inih to load the .ini file: https://github.com/benhoyt/inih
This library is under the New BSD license, thus I have added `LICENSE.txt` file into the `inih` directory.

## Technical info
I have created Settings.cpp and Settings.h to house the logic for loading settings.
I have also added `LoadSettings` in the `P2PConnection` struct to actually set the loaded settings and `SetDataFromSettingsIntoUI` in the `UI` files to load settings into the UI.

### Notable changes to the codebase
1. I have created `AllocAndShowConsole` in the `Utils` files to house the creation of console window (thus preventing duplicate code in two instances where we open console)
2. Added defined value `address_max_length` in the `UI.cpp` to minimize possible bugs (because I check whenever the length of the loaded address wouldn't exceed the address array max length)

## Postscript
I have tried to match your code style and your project setup. Hopefully, you won't find any difficulties with my code.
Also, if there would be any difficulties undestanding my PR, I do not mind explaining ^^

Feel free to suggest any changes!

### CODE REVIEW

**Comment by lilmayu** [comment]
I have noticed that you sometimes use tabs, so I have changed spaces to tabs in my changes to match the file tabs/spaces style.

Also I have noticed that there is file with windows-1250 encoding. If you have any problems with viewing my files, that are in UTF-8, I will gladly convert them to windows-1250.

File: Settings.cpp [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+	}
+
+	return this->ini_reader->GetInteger("Connection", "delay_compensation", -1);
+}
+
+int Settings::GetHostPort()
+{
+	if (this->ini_reader == nullptr)
+	{
+		return 18000; // Default port
```
Comment (williewillus) [inline Settings.cpp:95]: The default is 10800, not 18000:
https://github.com/RUEEE/th19p2vp/blob/c51e0b250c89c4e3ef75b825d8a1a2b439130637/Connection.cpp#L445

maybe put these in a separate constants file so you aren't repeating them 3 times (+ a fourth time in Connection.cpp)

File: Settings.cpp [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	}
+
+	return port;
+}
+
+int Settings::GetGuestPort()
+{
+	if (this->ini_reader == nullptr)
+	{
+		return 18001; // Default port
```
Comment (williewillus) [inline Settings.cpp:112]: same, this is wrong, the default is 10801, not 18001

**Review by williewillus — CHANGES_REQUESTED** [review]
noticed a couple typos

**Comment by RUEEE** [comment]
sorry but i've done this yesterday..
only a few bug left for test.
now it's updated that the patch will save/load settings from %appdata%/ShanghaiAlice/th19/p2vp_settings.ini

**Comment by lilmayu** [comment]
lol, nice. No worries.

**Pull request closed without merging**